hedyorg/hedy
# 💻 Adding a button for microbit

Adding a button that that transpiles the Hedy code to Micro-python code and then sends it to the user for downloading. 
The button is in class=hidden with id=saveMicrobitFilesContainer in templates/incl/editor-and-output.html
When the button is clicked sets the Microbit boolean to true and transpiles user code. The user code is saved in the Micro-bit folder. Then the saved code is converted to .hex and it is send to the user for downloading. At the end both files are deleted. 

Created @app.route for generating and downloading the files in app.py

**How to test**

Follow these steps to verify this PR works as intended:
Set the button to not be hidden. 
Click on the Run Micro bit button and a hex file should be downloaded automatically. Then the file can be uploaded [here](https://python.microbit.org/v/3) to check if the code is correct or directly saved to a Microbit via USB. 

**Checklist**
Done? Check if you have it all in place using this list: (mark with x if done)


If you're unsure about any of these, don't hesitate to ask. We're here to help!
